### ISSUE TITLE
ndarray.__legate_data_interface__ crashes

### ISSUE THREAD

**Issue by eddy16112** [nv-legate/legate.numpy#91]
I tried to access the legate data using`__legate_data_interface__`, but I am getting an error:
  File "./gemm.py", line 62, in run_gemm
    C_data = C.__legate_data_interface__
  File "/home/wwu/legate.numpy/legate/numpy/array.py", line 90, in __legate_data_interface__
    array = Array(arrow_type, [None, self._thunk])
  File "/home/wwu/legate.core/legate/core/legate.py", line 73, in __init__
    self._region = stores[1].storage[0]
  File "/home/wwu/legate.numpy/legate/numpy/eager.py", line 46, in storage
    return self.deferred.storage
  File "/home/wwu/legate.numpy/legate/numpy/deferred.py", line 169, in storage
    return (self.base.region, self.base.field.field_id)
AttributeError: 'Store' object has no attribute 'region'

Here is my code:
```
def run_gemm(N, I, ft):  # noqa: E741
    print("Problem Size:     M=" + str(N) + " N=" + str(N) + " K=" + str(N))
    print("Total Iterations: " + str(I))
    flops = total_flops(N, N, N)
    print("Total Flops:      " + str(flops / 1e9) + " GFLOPS/iter")
    space = total_space(N, N, N, ft)
    print("Total Size:       " + str(space / 1e6) + " MB")
    A, B, C = initialize(N, N, N, ft)
    # Compute some sums and check for NaNs to force synchronization
    # before we start the timing
    assert not math.isnan(np.sum(A))
    assert not math.isnan(np.sum(B))
    assert not math.isnan(np.sum(C))
    start = datetime.datetime.now()
    # Run for as many iterations as was requested
    #for idx in range(I):
    np.dot(A, B, out=C)
    C_data = C.__legate_data_interface__
```

**Comment by magnatelee** [comment]
@eddy16112 this should be fixed now. Can you pull and try again? You don't need to reinstall the code.

**Comment by eddy16112** [comment]
@magnatelee the error is gone, but if I change my code into 
```
np.dot(A, B, out=C)
C_data = C.__legate_data_interface__
np.dot(C, B, out=C)
```
then I am getting an error with the 2nd np.dot call
```
  File "./gemm.py", line 65, in run_gemm
    np.dot(C, B, out=C)
  File "/home/wwu/legate.numpy/legate/numpy/module.py", line 417, in dot
    return a_array.dot(b, out=out, stacklevel=2)
  File "/home/wwu/legate.numpy/legate/numpy/array.py", line 1057, in dot
    out._thunk.dot(
  File "/home/wwu/legate.numpy/legate/numpy/eager.py", line 314, in dot
    self.deferred.dot(rhs1, rhs2, stacklevel=(stacklevel + 1))
  File "/home/wwu/legate.numpy/legate/numpy/deferred.py", line 82, in wrapper
    result = func(*args, **kwargs)
  File "/home/wwu/legate.numpy/legate/numpy/deferred.py", line 68, in wrapper
    return func(*args, **kwargs)
  File "/home/wwu/legate.numpy/legate/numpy/deferred.py", line 98, in wrapper
    return func(*args, **kwargs)
  File "/home/wwu/legate.numpy/legate/numpy/deferred.py", line 994, in dot
    task.execute()
  File "/home/wwu/legate.core/legate/core/operation.py", line 124, in execute
    self._context.runtime.submit(self)
  File "/home/wwu/legate.core/legate/core/runtime.py", line 873, in submit
    self._schedule(ops)
  File "/home/wwu/legate.core/legate/core/runtime.py", line 866, in _schedule
    op.launch(strategy)
  File "/home/wwu/legate.core/legate/core/operation.py", line 183, in launch
    strategy.launch(launcher)
  File "/home/wwu/legate.core/legate/core/solver.py", line 126, in launch
    launcher.execute_single()
  File "/home/wwu/legate.core/legate/core/launcher.py", line 722, in execute_single
    result = self._context.dispatch(self.build_single_task(argbuf))
  File "/home/wwu/legate.core/legate/core/launcher.py", line 675, in build_single_task
    self._req_analyzer.analyze_requirements()
  File "/home/wwu/legate.core/legate/core/launcher.py", line 444, in analyze_requirements
    perm_map = field_set.coalesce()
  File "/home/wwu/legate.core/legate/core/launcher.py", line 404, in coalesce
    proj_infos = proj_set.coalesce()
  File "/home/wwu/legate.core/legate/core/launcher.py", line 374, in coalesce
    raise ValueError(
ValueError: Interfering requirements found: {(<legate.core.launcher.Broadcast object at 0x2af08b5674c0>, 0, 0), (<legate.core.launcher.Broadcast object at 0x2af08b567880>, 0, 0)}
```

If I remove the call to `C.__legate_data_interface__`, then it works fine. 
Any ideas?

**Comment by magnatelee** [comment]
I need a reproducer for this. Can you make one for me?

**Comment by magnatelee** [comment]
Nevermind. I just reproduced it.

**Comment by magnatelee** [comment]
@eddy16112 This should be fixed now. Can you pull and try again?

As noted in [this comment](https://github.com/nv-legate/legate.numpy/issues/57#issuecomment-915673265), the same issue can happen for any "in-place" operation (i.e., whose output can be redirected to one of the inputs). We will soon address all instances of this issue.

**Comment by eddy16112** [comment]
@magnatelee it works for me, thanks
